src/attribute.rs

```diff
@@ -418,7 +434,7 @@ impl<'n, 'f> NtfsAttribute<'n, 'f> {
             });
         }
 
-        let end = start as u32 + self.resident_value_length();
+        let end = u32::from(start).saturating_add(self.resident_value_length());
```

Comment:
Shouldn't this rather be a `checked_add`, bailing out early if `checked_add` returns that the addition overflows?
I don't like a `saturating_add`, because the sum may eventually be different from what was expected after adding two numbers.

Reply:
Yes I can agree, just did not know what to do with this one, I just made the 'easier' fix waiting for a discussion on this one and you did choose the better option with the whole context 👍🏼